### PR TITLE
[ros2] ros2 system status monitor

### PR DESCRIPTION
**Pull request by seanyen**
* `ROS2 System Status Webview`: use `xmlrpc` client to mimic [`ros2cli`](https://github.com/ros2/ros2cli/blob/master/ros2cli/ros2cli/daemon/__init__.py) behavior and query the available ROS2 system status. 
* `ROS2 StatusBarItem`:  replace `ros2 daemon status` CLI call with `xmlrpc` method call to query the system readiness, which reduces the overall CPU usage.

### CODE REVIEW

File: src/ros/ros2/ros2-monitor.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Andrew Short. All rights reserved.
```
Comment (kejxu) [inline src/ros/ros2/ros2-monitor.ts]: nit: please consider using Microsoft copyright header

File: src/ros/ros2/ros2-monitor.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            this.client.methodCall(method, [...args], (err, val) => {
+                if (err) {
+                    reject(err);
+                } else {
+                    resolve(val);
+                }
+            });
+        });
+    }
+}
```
Comment (kejxu) [inline src/ros/ros2/ros2-monitor.ts]: nit: add a new line after this

File: src/ros/ros2/ros2-monitor.ts (hunk 1 above)
Comment (seanyen) [reply src/ros/ros2/ros2-monitor.ts]: Good catch. Fixed it.

File: src/ros/ros2/ros2-monitor.ts (hunk 2 above)
Comment (seanyen) [reply src/ros/ros2/ros2-monitor.ts]: Fixed.

**Pull request merged**